src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -177,6 +221,47 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 navigationDocumentId: interfaceDocumentId);
         }
 
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToSameFileAsync(
+            Solution solution, INamedTypeSymbol extractedInterfaceSymbol, SyntaxNode typeNode, ProjectId projectId,
+            DocumentId documentId, INamedTypeSymbol typeToExtractFrom, IEnumerable<ISymbol> includedMembers, CancellationToken cancellationToken)
+        {
+            var symbolToDeclarationAnnotationMap = CreateSymbolToDeclarationAnnotationMap(
+                includedMembers,
+                ref solution,
+                out var documentIds,
+                typeNode,
+                out var typeNodeSyntaxAnnotation,
+                cancellationToken);
+
+            var document = solution.GetDocument(documentId);
+            var originalRoot = document.GetSyntaxRootSynchronously(cancellationToken);
```

Comment:
you are in async method? so, might just call GetSyntaxRootAsync? you actually call that below.

Reply:
I am not sure why this is resolved? this is private method, so I am not sure why it is using synchronous version of it rather than call async version and the top caller doing WaitAndGetResult ?